src/GraphQL.SystemTextJson/StringExtensions.cs

```diff
@@ -0,0 +1,43 @@
+using System.Collections.Generic;
+using System.Text.Json;
+
+namespace GraphQL.SystemTextJson
+{
+    public static class StringExtensions
+    {
+        private static readonly JsonSerializerOptions _jsonOptions = new JsonSerializerOptions
+        {
+            PropertyNamingPolicy = JsonNamingPolicy.CamelCase,
+            Converters =
+            {
+                new ObjectDictionaryConverter()
+            }
+        };
+
+        /// <summary>
+        /// Converts a JSON-formatted string into a dictionary.
+        /// </summary>
+        /// <param name="json">A JSON formatted string.</param>
+        /// <returns>Inputs.</returns>
+        public static Inputs ToInputs(this string json)
+            => json?.ToDictionary().ToInputs();
+
+        /// <summary>
+        /// Converts a JSON-formatted string into a dictionary of objects of their actual type.
+        /// </summary>
+        /// <param name="json">The json.</param>
+        /// <returns>Dictionary.</returns>
+        public static Dictionary<string, object> ToDictionary(this string json)
+            => JsonSerializer.Deserialize<Dictionary<string, object>>(json, _jsonOptions);
+
+        /// <summary>
+        /// Creates an <see cref="ExecutionResult"/> with it's <see cref="ExecutionResult.Data" />
+        /// property set to the strongly-typed representation of <paramref name="json"/>.
+        /// </summary>
+        /// <param name="json">A json representation of the <see cref="ExecutionResult.Data"/> to be set.</param>
+        /// <param name="errors">Any errors.</param>
+        /// <returns>ExecutionResult.</returns>
+        public static ExecutionResult ToExecutionResult(this string json, ExecutionErrors errors = null)
```

Comment:
But is this method generally needed somewhere other than tests? Will it be used by the server project?

Reply:
Nope. I can move it somewhere else if we have a common place between Harness.Tests and .Tests projects.

Reply:
Should I create a GraphQL.Tests.Core project? Essentially there was a tonne of methods before called `CreateQueryResult` that all used JObject directly, so I was trying to consolidate that.

Reply:
Let's use shared source code and not make new test projects.

Reply:
There's no common projects between those two other than i guess GraphQL.csproj, but it doesn't make sense in here since it's purely for unit tests. Anywhere else you think it should go? Or should I just copy paste it into both those projects and move on?

Reply:
I mean that shared code for tests stay in Test project (`shared` folder) and other projects use it as link.
```
  <ItemGroup>
    <Compile Include="..\GraphQL.Tests\Shared\*.cs" Link="Shared\*.cs" />
  </ItemGroup>
```

This solves copy-paste problem without creating new projects.